pkgs/development/compilers/uasm/default.nix

```diff
@@ -14,9 +20,23 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-CIbHPKJa60SyJeFgF1Tux7RfJZBChhUVXR7HGa+gCtQ=";
   };
 
+  patches = lib.optionals stdenv.isDarwin [
```

Comment:
This patch is not touching linux files so making it not conditional doesn't break anything. Making patches conditional increases the chance that with the next update the patch is forgotten when it no longer applies.

Reply:
I expect this patch to be removed on the next release though.

Reply:
Thats great but if someone else updates uasm and is not on Mac the chance to miss this are pretty realistic especially when using nix-update.

Reply:
Sure. If this happens we fix it when it happens.

I don't think we should bother opening another PR just to fix a possible issue.

Reply:
> Sure. If this happens we fix it when it happens.

Sure but most likely it will only be discovered X days/months later which puts unnecessary extra maintenance work onto us.



> Anyway, I will try to avoid doing this in the future (even if I don't like applying random patches unless strictly necessary).

That would be great!

Reply:
> Sure but most likely it will only be discovered X days/months later which puts unnecessary extra maintenance work onto us.

I still think this wouldn't actually happen since we still have `{aarch64,x86_64}-darwin` running in CI. The only way I can see this happening is someone bumping this and ignoring the CI checks (that I know it happens sometimes, but we shouldn't).

Reply:
Also, let's say this patch does affect other platforms but is only necessary for Darwin? Should we still apply it globally, having the risk of introducing bugs on all platforms because of one platform (that is known to be problematic in `nixpkgs`)?

If not, them we come again to the same issue that if we don't want to break Darwin we need to open the PR and wait for the CI checks at least. This is why I much prefer to apply patches only when strictly necessary, because I think supporting Darwin is good to have, but not essential.

Reply:
Anyway, I think this discussion extended too much. Is there some other place that we could discuss those things?

Reply:
> ignoring the CI checks (that I know it happens sometimes, but we shouldn't).

If I am in a hurry I usually tend to more easily ignore darwin failures. 

> Also, let's say this patch does affect other platforms but is only necessary for Darwin? Should we still apply it globally, having the risk of introducing bugs on all platforms because of one platform (that is known to be problematic in `nixpkgs`)?

I think that depends on the case. Often patches are only for some platform specific code.

---


You can chat me up on matrix if you like.

Reply:
> If I am in a hurry I usually tend to more easily ignore darwin failures.

If you care about Darwin, I think you shouldn't. Any change is a possible breakage, so the only way to ensure that nothing is breaking is at least leaving the CI to run until it finishes.

If you don't care, it is fine. But I don't think that having non conditional patches will help this that much, specially in one package that probably nobody cares. I (as the maintainer) particularly prefer that this package was updated even if the Darwin build was broken (for example, via @r-ryantm bot), because I don't have another way to be notified about packages updates otherwise.

Reply:
> Was going to send you a Matrix message but I don't know your handle.

``@sandro:supersandro.de``